apps/emqx_utils/src/emqx_utils.erl

```diff
@@ -548,6 +549,19 @@ safe_to_existing_atom(Atom, _Encoding) when is_atom(Atom) ->
 safe_to_existing_atom(_Any, _Encoding) ->
     {error, invalid_type}.
 
+-spec tcp_keepalive_opts(string() | binary()) ->
+    [{keepalive, true} | {raw, non_neg_integer(), non_neg_integer(), binary()}].
+tcp_keepalive_opts(None) when None =:= "none"; None =:= <<"none">> ->
+    [];
+tcp_keepalive_opts(KeepAlive) ->
+    {Idle, Interval, Probes} = emqx_schema:parse_tcp_keepalive(KeepAlive),
```

Comment:
Q: Why not keep schema-related code in `emqx` app? There's now weird circular (runtime) dependency between `emqx_utils` and `emqx`.